ksql-engine/src/test/java/io/confluent/ksql/util/KafkaTopicClientTest.java

```diff
@@ -124,6 +129,58 @@ public void shouldNotFailIfTopicAlreadyExistsWhenCreating() throws InterruptedEx
     KafkaTopicClient kafkaTopicClient = new KafkaTopicClientImpl(adminClient);
     kafkaTopicClient.createTopic(topicName1, 1, (short)1, false);
     verify(adminClient);
+
+  }
+
+
+  // Compares two collections of 'kafka.clients.admin.NewTopic' objects to verify that they are
+  // the same. The NewTopic.equals method isn't implemented so it just does reference equality,
+  // which is not what we want in these tests.
+  class NewTopicComparator implements Comparator<Collection<NewTopic>> {
+
+    @Override
+    public int compare(Collection<NewTopic> topicCollection1,
+                       Collection<NewTopic> topicCollection2) {
+      if (topicCollection1.size() != topicCollection2.size()) {
+        return -1;
+      }
+
+      Iterator<NewTopic> firstIterator = topicCollection1.iterator();
+      Iterator<NewTopic> secondIterator = topicCollection2.iterator();
+
+      while (firstIterator.hasNext()) {
+        NewTopic topic1 = firstIterator.next();
+        NewTopic topic2 = secondIterator.next();
+        if (!(topic1.name().equals(topic2.name())
```

Comment:
That's a pretty hacky comparator impl! ;)

`Comparator`s are all about sorting collections, which is not what you want to achieve here. Why not just use a custom argument matcher that does what you want? e.g. 

```
private static Collection<NewTopic> singleNewTopic(final NewTopic expected) {
    class NewTopicsMatcher implements IArgumentMatcher {
      @SuppressWarnings("unchecked")
      @Override
      public boolean matches(final Object argument) {
        final Collection<NewTopic> newTopics = (Collection<NewTopic>) argument;
        if (newTopics.size() != 1) {
          return false;
        }

        final NewTopic actual = newTopics.iterator().next();
        return Objects.equals(actual.name(), expected.name())
               && Objects.equals(actual.replicationFactor(), expected.replicationFactor())
               && Objects.equals(actual.numPartitions(), expected.numPartitions())
               && Objects.equals(actual.configs(), expected.configs());
      }

      @Override
      public void appendTo(final StringBuffer buffer) {
        buffer.append("{NewTopic").append(expected).append("}");
      }
    }

    EasyMock.reportMatcher(new NewTopicsMatcher());
    return null;
  }
```

Then you can change the test to:

```
  @Test
  public void shouldSetTopicCleanupPolicyToCompact()  {
    AdminClient adminClient = mock(AdminClient.class);
    expect(adminClient.describeCluster()).andReturn(getDescribeClusterResult());
    expect(adminClient.listTopics()).andReturn(getEmptyListTopicResult());
    expect(adminClient.describeConfigs(anyObject())).andReturn(getDescribeConfigsResult());

    // Verify that the new topic configuration being passed to the admin client is what we expect.
    NewTopic newTopic = new NewTopic(topicName1, 1, (short) 1);
    newTopic.configs(Collections.singletonMap("cleanup.policy", "compact"));
  
    expect(adminClient.createTopics(singleNewTopic(newTopic))).andReturn(getCreateTopicsResult());
    replay(adminClient);
    
    KafkaTopicClient kafkaTopicClient = new KafkaTopicClientImpl(adminClient);
    kafkaTopicClient.createTopic(topicName1, 1, (short) 1, true);
    verify(adminClient);
  }
```

Reply:
Thanks @big-andy-coates . One of the motivations for doing this was to learn about how to do this nicely with EasyMock. Thanks for the tip! I implemented your changes.